NomadAudio/src/PatternPlaybackEngine.cpp

```
@@ -74,9 +88,43 @@ void PatternPlaybackEngine::refillWindow(uint64_t currentFrame, int sampleRate,
+                    break;
+                }
+            }
+            
+            if (!isSorted) {
+                // Make a mutable copy and sort it
+                auto notesCopy = midi.notes;
+                std::sort(notesCopy.begin(), notesCopy.end(), 
+                    [](const MidiNote& a, const MidiNote& b) { return a.startBeat < b.startBeat; });
+                
+                // Replace the pattern's notes with sorted version
+                const_cast<MidiPayload&>(midi).notes = std::move(notesCopy);
```

Cppcheck (2.19.0)

[style] The function 'isOpen' is never used.

(unusedFunction)